--- a/source/source_esolver/test/CMakeLists.txt	2026-06-24 01:42:15.679803353 +0800
+++ b/source/source_esolver/test/CMakeLists.txt	2026-06-24 01:42:16.927803342 +0800
@@ -3,8 +3,10 @@
 
 install(DIRECTORY support DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
 
+if(DEFINED DeePMD_DIR)
 AddTest(
   TARGET MODULE_ESOLVER_esolver_dp_test
-  LIBS parameter ${math_libs} base device 
+  LIBS parameter ${math_libs} base device DeePMD::deepmd_c
   SOURCES esolver_dp_test.cpp ../esolver_dp.cpp ../../source_io/module_output/cif_io.cpp ../../source_io/module_output/output_log.cpp
 )
+endif()
